### PR TITLE
Documented parse_expr in the sympify() docstring

### LINKED ISSUE #21207: Document parse_expr in the sympify() docstring
In the example below, sympy raises a fuss because it is missing a multiplier between `2` and `x`.
It looks like it attempts to sympify the expression `'could not parse '2x'`, which is strange and confusing.

```python
import sympy as sp
sp.sympify("2x+1")

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
ValueError: Error from parse_expr with transformed code: "Integer (2 )Symbol ('x' )"

The above exception was the direct cause of the following exception:

SyntaxError                               Traceback (most recent call last)
c:\users\thomasaar\documents\github\sympy\sympy\core\sympify.py in sympify(a, locals, convert_xor, strict, rational, evaluate)
    478         a = a.replace('\n', '')
--> 479         expr = parse_expr(a, local_dict=locals, transformations=transformations, evaluate=evaluate)
    480     except (TokenError, SyntaxError) as exc:

c:\users\thomasaar\documents\github\sympy\sympy\parsing\sympy_parser.py in parse_expr(s, local_dict, transformations, global_dict, evaluate)
   1016     except Exception as e:
-> 1017         raise e from ValueError(f"Error from parse_expr with transformed code: {code!r}")
   1018 

c:\users\thomasaar\documents\github\sympy\sympy\parsing\sympy_parser.py in parse_expr(s, local_dict, transformations, global_dict, evaluate)
   1014     try:
-> 1015         return eval_expr(code, local_dict, global_dict)
   1016     except Exception as e:

c:\users\thomasaar\documents\github\sympy\sympy\parsing\sympy_parser.py in eval_expr(code, local_dict, global_dict)
    908     """
--> 909     expr = eval(
    910         code, global_dict, local_dict)  # take local objects in preference

SyntaxError: invalid syntax (<string>, line 1)

During handling of the above exception, another exception occurred:

SympifyError                              Traceback (most recent call last)
<ipython-input-37-d8899059adbf> in <module>
      1 import sympy as sp
----> 2 sp.sympify("2x")

c:\users\thomasaar\documents\github\sympy\sympy\core\sympify.py in sympify(a, locals, convert_xor, strict, rational, evaluate)
    479         expr = parse_expr(a, local_dict=locals, transformations=transformations, evaluate=evaluate)
    480     except (TokenError, SyntaxError) as exc:
--> 481         raise SympifyError('could not parse %r' % a, exc)
    482 
    483     return expr

SympifyError: Sympify of expression 'could not parse '2x'' failed, because of exception being raised:
SyntaxError: invalid syntax (<string>, line 1)
```

### PR DESCRIPTION
**Pull request by Jatin2020-24**
#### References to other Issues or PRs
Fixes : #21207

#### Brief description of what is fixed or changed
Documented `parse_expr` in the sympify() docstring.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by asmeurer** [comment]
We should also add a "see also" section.

**Comment by Jatin2020-24** [comment]
@asmeurer Why sphinx test is failing in last commit. Please suggest some way so these tests can be passed.

**Comment by sidhu1012** [comment]
@Jatin2020-24 The problem is in the docstring you added.

**Comment by Jatin2020-24** [comment]
@asmeurer I have added "see also" section. Please review this PR.

**Comment by Jatin2020-24** [comment]
> @Jatin2020-24 The problem is in the docstring you added.

@sidhu1012 Thanks for your guidance.

**Comment by Jatin2020-24** [comment]
@asmeurer Can this PR be merged.

File: sympy/core/sympify.py [hunk 1]
```diff
@@ -333,6 +346,11 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
         their SymPy equivalents. If True the expression will be evaluated
         and the result will be returned.
 
+    See Also
+    ========
+
+    parse_expr
```
Comment (smichr) [inline sympy/core/sympify.py:352]: I think this needs to have the full path if it is not in the same file that it refers to: `sympy.parsing.sympy_parser.parse_expr`?

Comment (oscarbenjamin) [reply sympy/core/sympify.py:352]: I haven't really got my head round how it works but it doesn't seem like "see also" uses the same namespace as other sphinx directives.

**Pull request closed without merging**